src/Adapter/Order/Refund/OrderSlipCreator.php

```diff
@@ -138,13 +147,184 @@ public function createOrderSlip(
+                }
+            }
+
+            $product_tax_incl_line = Tools::ps_round($tax_calculator->{$add_or_remove . 'Taxes'}($price) * $quantity, $precision);
+
+            switch ($this->configuration->get('PS_ROUND_TYPE')) {
+                case Order::ROUND_ITEM:
+                    $product_tax_incl = Tools::ps_round($tax_calculator->{$add_or_remove . 'Taxes'}($price), $precision) * $quantity;
+                    $total_products[$id_tax_rules_group] += $product_tax_incl;
+
+                    break;
+                case Order::ROUND_LINE:
+                    $product_tax_incl = $product_tax_incl_line;
+                    $total_products[$id_tax_rules_group] += $product_tax_incl;
+
+                    break;
+                case Order::ROUND_TOTAL:
+                    $product_tax_incl = $product_tax_incl_line;
+                    $total_products[$id_tax_rules_group . '_' . $id_address] += $price * $quantity;
+
+                    break;
+            }
+
+            $product['unit_price_tax_' . $inc_or_ex_1] = $price;
+            $product['unit_price_tax_' . $inc_or_ex_2] = Tools::ps_round($tax_calculator->{$add_or_remove . 'Taxes'}($price), $precision);
+            $product['total_price_tax_' . $inc_or_ex_1] = Tools::ps_round($price * $quantity, $precision);
+            $product['total_price_tax_' . $inc_or_ex_2] = Tools::ps_round($product_tax_incl, $precision);
+        }
+
+        unset($product);
+
+        foreach ($total_products as $key => $price) {
+            if ($this->configuration->get('PS_ROUND_TYPE') == Order::ROUND_TOTAL) {
+                $tmp = explode('_', $key);
+                $address = Address::initialize((int) $tmp[1], true);
+                $tax_calculator = TaxManagerFactory::getManager($address, $tmp[0])->getTaxCalculator();
+                $order_slip->{'total_products_tax_' . $inc_or_ex_2} += Tools::ps_round($tax_calculator->{$add_or_remove . 'Taxes'}($price), $precision);
+            } else {
+                $order_slip->{'total_products_tax_' . $inc_or_ex_2} += $price;
+            }
+        }
+
+        $order_slip->{'total_products_tax_' . $inc_or_ex_2} -= $amount && !$amount_choosen ? $amount : 0;
+        $order_slip->amount = $amount_choosen ? $amount : $order_slip->{'total_products_tax_' . $inc_or_ex_1};
+        $order_slip->shipping_cost_amount = $order_slip->total_shipping_tax_incl;
+
+        if ((float) $amount && !$amount_choosen) {
+            $order_slip->order_slip_type = 1;
+        }
+        if (((float) $amount && $amount_choosen) || $order_slip->shipping_cost_amount > 0) {
+            $order_slip->order_slip_type = 2;
+        }
+
+        if (!$order_slip->add()) {
+            return false;
+        }
+
+        $res = true;
+
+        foreach ($product_list as $product) {
+            $res &= $this->addProductOrderSlip((int) $order_slip->id, $product);
+        }
+
+        return $res;
```

Comment:
```suggestion
        return (bool) $res;
```
Must be cast in bool

Reply:
Mmm, it's already adding boolean only anyway But if we want to be sure I prefer to add it explicitly on the `addProductOrderSlip` method

Reply:
Because of bitwise, it's not a boolean, it's an integer :) 
https://www.php.net/manual/en/language.operators.bitwise.php

If you want to keep the integer, you need to change the php doc block